src/js/wall/Reset.jsx

```diff
@@ -14,13 +14,22 @@ export class Reset extends React.Component {
         };
     }
 
+    componentDidMount() {
+        this.setState({
+            resetCode: this.props.resetCode,
```

Comment:
Did you figure out why the reset code get forgotten after the first request? Not totally against this soln if it makes the most sense, just seems strange that the code persists for only one request